Library/Homebrew/os/linux/elf.rb

```diff
@@ -87,15 +87,18 @@ def with_interpreter?
   end
 
   def dynamic_elf?
-    return @dynamic_elf if defined? @dynamic_elf
+    return @dynamic_elf if defined?(@dynamic_elf)
 
-    @dynamic_elf = if which "readelf"
-      Utils.popen_read("readelf", "-l", to_path).include?(" DYNAMIC ")
-    elsif which "file"
-      !Utils.popen_read("file", "-L", "-b", to_path)[/dynamic|shared/].nil?
-    else
-      raise "Please install either readelf (from binutils) or file."
-    end
+    @dynamic_elf =
+      if HOMEBREW_PATCHELF_RB
+        patcher.instance_variable_get("@elf").segment_by_type(:DYNAMIC).present?
```

Comment:
```suggestion
        patcher.elf.segment_by_type(:DYNAMIC).present?
```
Why is `.instance_variable_get("@elf")` needed here? Is there an `elf` method that returns `@elf`? If not, could you open a PR upstream to add this method? If we do need `instance_variable_get`, let's use a symbol `.instance_variable_get(:@elf)`.

Reply:
@sjackman i'm breaking the abstraction here, as patchelf is not supposed to expose how it implements the patching method.
on the practical side, i do this so i don't have to create 2 objects, one PatchELF which will be used for patching and some general queries, and `ELFFile` object. 

A more proper patch to upstream will be to make `PatchELF::Patcher` accept  ELFFile as argument.

Reply:
```suggestion
        patcher.instance_variable_get(:@elf).segment_by_type(:DYNAMIC).present?
```
I'm fine with this for now, but I'd like to eventually address this upstream with a PR.